pkg/koordlet/metricsadvisor/collectors/beresource/be_resource_collector.go

```diff
@@ -14,33 +14,82 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package metricsadvisor
+package beresource
 
 import (
+	"os"
 	"time"
 
+	"go.uber.org/atomic"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
 	apiext "github.com/koordinator-sh/koordinator/apis/extension"
 	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metricsadvisor/framework"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/resourceexecutor"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer"
 	koordletutil "github.com/koordinator-sh/koordinator/pkg/koordlet/util"
 	"github.com/koordinator-sh/koordinator/pkg/util"
 )
 
-func (c *collector) collectBECPUResourceMetric() {
+type beResourceColelctor struct {
+	collectInterval time.Duration
+	started         *atomic.Bool
+	metricDB        metriccache.MetricCache
+	statesInformer  statesinformer.StatesInformer
+	cgroupReader    resourceexecutor.CgroupReader
+
+	lastBECPUStat framework.CPUStat
+}
+
+func New(opt *framework.Options) framework.Collector {
+	return &beResourceColelctor{
+		collectInterval: time.Duration(opt.Config.CollectResUsedIntervalSeconds) * time.Second,
+		started:         atomic.NewBool(false),
+		metricDB:        opt.MetricCache,
+		statesInformer:  opt.StatesInformer,
+		cgroupReader:    opt.CgroupReader,
+	}
+}
+
+func (b *beResourceColelctor) Enabled() bool {
+	return true
+}
+
+func (b *beResourceColelctor) Setup(s *framework.Context) {
+	return
+}
+
+func (b *beResourceColelctor) Run(stopCh <-chan struct{}) {
+	if !cache.WaitForCacheSync(stopCh, b.statesInformer.HasSynced) {
+		klog.Errorf("timed out waiting for states informer caches to sync")
+		// Koordlet exit because of statesInformer sync failed.
+		os.Exit(1)
```

Comment:
how about `klog.Fatalf`? or do we expect a fixed return code for this kind of failure